src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Editor/EditorSettingsChangedEventArgs.cs

```diff
@@ -5,17 +5,7 @@
 
 namespace Microsoft.CodeAnalysis.Razor.Editor;
 
-public sealed class EditorSettingsChangedEventArgs : EventArgs
+internal sealed class EditorSettingsChangedEventArgs(EditorSettings settings) : EventArgs
 {
-    public EditorSettingsChangedEventArgs(EditorSettings settings)
-    {
-        if (settings is null)
-        {
-            throw new ArgumentNullException(nameof(settings));
-        }
-
-        Settings = settings;
-    }
-
-    public EditorSettings Settings { get; }
+    public EditorSettings Settings { get; } = settings ?? throw new ArgumentNullException(nameof(settings));
```

Comment:
Not for this PR, but given we can now categorically say we have no public API, and we're (mostly) nullable enabled, is there any valid reason for keeping all of our argument validation?

Reply:
The only reason that gives me pause is that it's a potentially disruptive change. I've seen a fair amount of test code passing `null` where it shouldn't.

Aside: @ryzngard, this is a potential reason in favor of enabling nullability in tests -- to spot these concerns quickly.